src/python/pants/backend/go/module.py

```diff
@@ -249,12 +250,21 @@ async def download_external_module(
     source_path = absolute_source_path[gopath_index:]
 
     source_digest = await Get(
-        Digest, DigestSubset(result.output_digest, PathGlobs([f"{source_path}/**"]))
+        Digest,
+        DigestSubset(
+            result.output_digest,
+            PathGlobs(
+                [f"{source_path}/**"],
+                glob_match_error_behavior=GlobMatchErrorBehavior.error,
+                description_of_origin=(
+                    f"Attempted to download external Go module {request.path}@{request.version}, "
+                    "but the download was empty."
```

Comment:
The arg name isn't very descriptive and requires reading the docstring, which is annoying. Should not be a complete sentence because this gets combined w/ Rust error message. Should probably be:

> the `DownloadedExternalModule` for `{request.path}@{request.version}`

The message would then read:

> Unmatched glob from the `DownloadedExternalModule` for `{request.path}@{request.version}`: bad/glob**

Reply:
Would it be better to refer to the request type instead (i.e., `DownloadExternalModuleRequest`)?

Reply:
Fixed.